### PR TITLE
Add senlin model

### PR DESCRIPTION
**Pull request by L-I-O-N**
As openstack mitaka version add a new core project which is called Senlin. According to the basic framework, I Add a new module to implements Senlin RESRful API.And I hava tested all the basic functions of this module. #684

### CODE REVIEW

**Comment by auhlig** [comment]
Why are you introducing the ActionID ? Shouldn't the actions id be an attribute in the, already existing, Action?

**Comment by L-I-O-N** [comment]
This is because the return of the action and the acquisition of the action format is different.
{
    "action": {
        "action": "CLUSTER_SCALE_IN",
        "cause": "RPC Request",
        "created_at": "2016-06-15T02:42:38.000000",
        ........
    }
}
and
{
    "action": "40a436b1-28d1-4de6-b2c3-0a34f478e2c9"
}

**Comment by L-I-O-N** [comment]
@gondor Any question? This matter has been on hold for almost a month.

**Comment by vinodborole** [comment]
@L-I-O-N Apologies for the delay and thank you for your patience.

@auhlig @gondor  LGTM

**Comment by L-I-O-N** [comment]
Thanks for your support. @vinodborole. Our projects have used this SDK,so there is no obvious problems on function.

File: core/src/main/java/org/openstack4j/openstack/senlin/domain/SenlinAction.java [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+		return target;
+	}
+
+	@Override
+	public Integer getTimeout() {
+		return timeout;
+	}
+
+	@Override
+	public String toString() {
```
Comment (auhlig) [inline core/src/main/java/org/openstack4j/openstack/senlin/domain/SenlinAction.java]: Why not use something like 
```
public String toString() {
  return Objects.toStringHelper(this).omitNullValues()
                .add("action", action)
                .add("cause", cause)
                // ...
                .toString();
    }
```

**Comment by auhlig** [comment]
I hesitate to follow your style. 
Just one example: Looking at [SenlinClusterCreate.java](https://github.com/L-I-O-N/openstack4j/blob/1624fcfca167851c6c67d3ba2469357588504815/core/src/main/java/org/openstack4j/openstack/senlin/domain/SenlinClusterCreate.java).
Comparing your implementation to other parts of os4j I'd suggest to create a class with @JsonRootName("cluster") and attributes like name, max_size,... from the response instead of creating a cluster-map and adding all those there.

**Pull request merged**